packages/next/src/server/base-server.ts

```diff
@@ -2681,15 +2693,36 @@ export default abstract class Server<
             // https://github.com/vercel/next.js/blob/df7cbd904c3bd85f399d1ce90680c0ecf92d2752/packages/next/server/render.tsx#L947-L952
             renderOpts.nextFontManifest = this.nextFontManifest
 
-            // Call the built-in render method on the module.
-            result = await module.render(req, res, {
+            const context: AppPageRouteHandlerContext = {
               page: is404Page ? '/404' : pathname,
               params: opts.params,
               query,
               fallbackRouteParams,
               renderOpts,
               serverComponentsHmrCache: this.getServerComponentsHmrCache(),
-            })
+            }
+
+            // If we're in dev, and this isn't s prefetch or a server action,
+            // we should seed the resume data cache.
+            if (
+              this.nextConfig.experimental.dynamicIO &&
+              this.renderOpts.dev &&
+              !isPrefetchRSCRequest &&
+              !isServerAction
+            ) {
+              renderOpts.isDevWarmup = true
+              const warmup = await module.render(req, res, context)
+              renderOpts.isDevWarmup = false
+
+              // If the warmup is successful, we should use the resume data
+              // cache from the warmup.
+              if (warmup.metadata.type === 'app') {
+                renderOpts.resumeDataCache = warmup.metadata.resumeDataCache
+              }
```

Comment:
If we get to the point where we want to implement this we could link this into the existing incremental cache system rather than relying on this timing based method that was here previously.